ledger/sandbox/src/main/resources/db/migration/postgres/V34__Parties_is_local.sql

```diff
@@ -0,0 +1,11 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+---------------------------------------------------------------------------------------------------
+-- V34: Add is_local column to parties table.
+--
+-- This allows parties to be marked as to whether they were added locally as determined by whether
+-- the specified participant_id matched the local participant_id.
+---------------------------------------------------------------------------------------------------
+
+ALTER TABLE parties ADD COLUMN is_local bool not null default true;
```

Comment:
I guess you are right. Right now, all parties appear as local, so the change would have only applied to new parties. We can fix the existing parties by doing the following. Keep above change, but extend it with:
```
UPDATE parties SET is_local = false WHERE party IN (SELECT party FROM party_entries WHERE is_local = false);
ALTER TABLE parties CHANGE COLUMN is_local is_local bool not null;
```
That was actually also the issue: we have two party tables.

Reply:
`parties` are allocated parties, `party_entries` represents the history of allocations. We have two because the latter keeps track of ongoing allocations (since they have to happen asynchronously) while the former tracks the state (more or less akin to contracts and transactions/events).

Reply:
Why not add another column to party_entries active boolean and an appropriate index? Then the "current state" is given by all the party_entries that are active and the rest captures the process and the history? -1 table.